Logger/BaseLoggerMixins.cs

```diff
@@ -1,6 +1,35 @@
-﻿namespace Logger;
 
-public static class BaseLoggerMixins
+using System;
+using System.Runtime.CompilerServices;
+using static Logger.BaseLogger;
+
+namespace Logger
 {
-    
+    public static class BaseLoggerMixins
+    {
+        public static void Error(this BaseLogger log, string message, params object[] args)
+        {
+            if (log == null)
```

Comment:
```suggestion
            if (log is null)
```
Could use `is` keyword to match on null

Also, you could pull this out into a private static helper method that takes a BaseLogger and throws if it's null, so you don't have to write it out an additional 3 times